test/krew_test.go

```diff
@@ -51,3 +177,10 @@ func skipShort(t *testing.T) {
 		t.Skip("skipping integration test")
 	}
 }
+
+func lines(in []byte) []string {
+	if string(in) == "" {
+		return nil
+	}
+	return strings.Split(string(in), "\n")
```

Comment:
ideally you should do `strings.TrimRight(string(in))` so that you avoid taking trailing `\n` which almost  all commands will have.

Reply:
with this,  you don't   need the =="" check,

Reply:
Unfortunately, the `==""` is still required. For `krew list` I need the list of results to grow by one if a plugin is installed. However, when starting from none (resulting in "\n") to one plugin ("whatever\n") would result in 1 entry each.

Reply:
hmm. I'm not sure if that's accurate.

Since in this case the stdout is not a tty, we don't print the tabular output.

I expect no output or at most `\n`. Both of which would return `[]string{}` (len=0).

When you install a plugin I expect this to return `pluginname\n`, if you split it by `\n` it should return  `[]string{"pluginnname"}` (len=1). 

Alternatively you can also test that case as `output.Contains("pluginname")`, which is also just fine.

Reply:
> I expect no output or at most `\n`. Both of which would return `[]string{}` (len=0).

That is unfortunately not the case. If the input string is empty or "\n", it will result in `[]string{""}`.